core/crypto/vrf/impl/vrf_provider_impl.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/vrf/impl/vrf_provider_impl.hpp"
+
+#include "common/outcome.hpp"
+
+namespace fc::crypto::vrf {
+
+  VRFProviderImpl::VRFProviderImpl(
+      std::shared_ptr<bls::BlsProvider> bls_provider)
+      : bls_provider_(std::move(bls_provider)) {}
```

Comment:
Maybe ASSERT it's not null, as we used to in kagome?

Reply:
added assertion